src/coreclr/vm/eetwain.cpp

```diff
@@ -1533,6 +1550,24 @@ bool EECodeManager::EnumGcRefs( PREGDISPLAY     pRD,
                         curOffs
                         );
 
+    if ((flags & ActiveStackFrame) != 0)
+    {
+        // CONSIDER: We can optimize this by remembering the need to adjust in IsSafePoint and propagating into here.
+        //           Or, better yet, maybe we should change the decoder to not require this adjustment.
+        //           The scenario that adjustment tries to handle (fallthrough into BB with random liveness)
+        //           does not seem possible.
+        if (!gcInfoDecoder.HasInterruptibleRanges())
+        {
+            gcInfoDecoder = GcInfoDecoder(
+                gcInfoToken,
+                GcInfoDecoderFlags(DECODE_GC_LIFETIMES | DECODE_SECURITY_OBJECT | DECODE_VARARG),
+                curOffs - 1
```

Comment:
The code above already alters `curOffs` by decrementing it by one in some cases, e.g. https://github.com/dotnet/runtime/blob/ce3435737bbe46014aad148bf9cfeec1b80bebc3/src/coreclr/vm/eetwain.cpp#L1460-L1465. Would we ever end up double-subtracting? Should this `-1` logic be above, where the other "alter curOffs" logic is? You should at least add `LOG` output (like in that link) stating that we're now looking for GC info at a new offset (and what that offset is).

IMO, you should also document what `- 1` means here, and why it's ok to potentially point in the middle of an instruction.

Reply:
The "-1" is needed for historical reasons. Unlike interruptible ranges, which are recorded in terms of actual offsets, the safepoints are recorded with -1 skew. 
Thus if we have a leaf, but not fully-interruptible method, we need to do this -1 adjustment.

This is confusing and I'd like to remove this adjustment. However we are not changing the actual GCInfo format in this change, thus I need to adjust. 

Other/existing places have comments about the adjustment. I will add something here.

> Would we ever end up double-subtracting?

That would lead to lots of failures. 
I think the logic is correct, but as I said, I do agree that having to do this adjustment is confusing.

Reply:
Here is an example when this `-1` adjustment gets us into trouble. - https://github.com/dotnet/runtime/issues/86273

I'd like to get rid of the adjustment, when/if this gets in.

Reply:
> This is confusing and I'd like to remove this adjustment. However we want to stay backwards-compatible in this PR, thus I need to adjust

Note that David is planning to reset R2R band in #100310 that allows you to fix this up without worrying about backward-compatibility.